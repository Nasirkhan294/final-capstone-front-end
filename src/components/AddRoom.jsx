import React, { useEffect } from 'react';
import {
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  Checkbox,
  Typography,
  Input,
  Button,
  Textarea,
} from '@material-tailwind/react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import {
  ErrorMessage, Formik, Form, Field,
} from 'formik';
import * as Yup from 'yup';
import {
  addRoom,
  selectRoom,
  selectRoomMessage,
  selectRoomStatus,
} from '../redux/Room/roomSlice';
import useToken from '../redux/Auth/useToken';
import Alert from './Alert';
import { Spinner } from './Loader';

const AddRoom = () => {
  const defaultImg = 'https://www.fluttercampus.com/img/4by3.webp';
  const message = useSelector(selectRoomMessage);
  const status = useSelector(selectRoomStatus);
  const availableRooms = useSelector(selectRoom);
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const isTokenSet = useToken();

  const initialValues = {
    location: '',
    description: '',
    price: '',
    capacity: '',
    image: defaultImg,
    available: true,
  };

  const RoomSchema = Yup.object().shape({
    location: Yup.string()
      .min(4, 'Too Short!')
      .max(250, 'Too Long!')
      .matches(
        /^(?=.{4,250}$)[a-zA-Z0-9 ]+$/,
        'Location should have at least 4 characters and contain only letters, numbers, and spaces!',
      )
      .required('Location is required!'),
    description: Yup.string()
      .required('Description is required!')
      .min(5, 'Too Short!')
      .max(500, 'Too Long!'),
    price: Yup.number()
      .required('Price is required!')
      .positive('Please enter only positive numbers!'),
    capacity: Yup.number()
      .required('Price is required!')
      .positive('Please enter only positive numbers!'),
    image: Yup.string()
      .url('Invalid URL')
      .required('Image is required!')
      .matches(
        /\.(gif|jpe?g|tiff?|png|webp|bmp)$/i,
        'Should end with gif, jpeg, tiff, png, webp, bmp, or jpg',
      ),
  });

  document.title = 'Room Rental | Add Room';

  const handleAddRoom = (room) => {
    const roomdata = {
      location: room.location,
      capacity: room.capacity,
      price: room.price,
      floor: 'N/A',
      description: room.description,
      img: room.image,
    };
    dispatch(addRoom(roomdata));
    navigate('/');
  };

  const checkAuthUser = () => {
    if (!isTokenSet) navigate('/login');
  };

  useEffect(() => {
    checkAuthUser();
  }, [message, isTokenSet, availableRooms]);
  return (
    <>
      {status === 'failed' && <Alert message={message} />}
      <Card className="mt-8 mb-64 max-w-[450px] mx-auto bg-white/90 backdrop-blur-md">
        <CardHeader
          variant="gradient"
          className="mb-4 grid h-28 place-items-center bg-orange-600/90 text-white backdrop-blur-md"
        >
          <Typography
            variant="h3"
            color="white"
            className="font-osans uppercase tracking-widest font-light"
          >
            Add a room
          </Typography>
        </CardHeader>
        <CardBody className="text-center">
          <Typography variant="h5" className="mb-2">
            Kind Note
          </Typography>
          <Typography className="text-center">
            Kindly fill in the form below to add a room to the list of available
            rooms.
          </Typography>
        </CardBody>
        <Formik
          initialValues={initialValues}
          onSubmit={handleAddRoom}
          validationSchema={RoomSchema}
        >
          {({
            errors, touched, isValid, dirty, values,
          }) => (
            <Form>
              <CardBody className="flex flex-col gap-4">
                <div>
                  <img src={values.image} alt="room-preview" />
                </div>
                <Field
                  as={Input}
                  name="location"
                  color="orange"
                  size="md"
                  label="Location"
                  error={Boolean(errors.location) && touched.location}
                />
                <ErrorMessage
                  name="location"
                  render={(msg) => (
                    <span className="text-xs text-gray-500">{msg}</span>
                  )}
                />
                <Field
                  as={Input}
                  name="image"
                  color="orange"
                  size="md"
                  label="Put a Room Image link"
                  error={Boolean(errors.image) && touched.image}
                />
                <ErrorMessage
                  name="image"
                  render={(msg) => (
                    <span className="text-xs text-gray-500">{msg}</span>
                  )}
                />
                <Field
                  as={Input}
                  name="price"
                  type="number"
                  color="orange"
                  size="lg"
                  label="Price"
                  error={Boolean(errors.price) && touched.price}
                />
                <ErrorMessage
                  name="price"
                  render={(msg) => (
                    <span className="text-xs text-gray-500">{msg}</span>
                  )}
                />
                <Field
                  as={Input}
                  name="capacity"
                  type="number"
                  color="orange"
                  size="lg"
                  label="capacity"
                  error={Boolean(errors.capacity) && touched.capacity}
                />
                <ErrorMessage
                  name="capacity"
                  render={(msg) => (
                    <span className="text-xs text-gray-500">{msg}</span>
                  )}
                />
                <Field
                  as={Textarea}
                  name="description"
                  color="orange"
                  label="Description"
                  size="md"
                  error={Boolean(errors.description) && touched.description}
                />
                <ErrorMessage
                  name="description"
                  render={(msg) => (
                    <span className="text-xs text-gray-500">{msg}</span>
                  )}
                />
                <Field
                  as={Checkbox}
                  name="available"
                  color="orange"
                  defaultChecked
                  label={values.available ? 'Available' : "Don't put in list"}
                />
              </CardBody>
              <CardFooter className="pt-0">
                <Button
                  type="submit"
                  color="orange"
                  variant="gradient"
                  fullWidth
                  className="capitalize flex justify-center items-center"
                  disabled={!isValid || !dirty}
                >
                  {status === 'loading' ? <Spinner /> : <span>Add Room</span>}
                </Button>
              </CardFooter>
            </Form>
          )}
        </Formik>
      </Card>
    </>
  );
};

export default AddRoom;
