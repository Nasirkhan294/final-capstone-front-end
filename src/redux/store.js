import { configureStore } from '@reduxjs/toolkit';
import authReducer from './Auth/authSlice';
import roomsReducer from './Room/roomSlice';
import reservationReducer from './Reservations/reservationsSlice';

// Redux store
const store = configureStore({
  reducer: {
    auth: authReducer,
    rooms: roomsReducer,
    bookings: reservationReducer,
  },
});

export default store;
