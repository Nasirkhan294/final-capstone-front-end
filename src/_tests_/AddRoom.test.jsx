import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/store';
import '@testing-library/jest-dom/';
import AddRoom from '../components/AddRoom';

describe('AddRoom component', () => {
    test('Should render the proper AddRoom component', () => {
      const addRoom = render(
        <React.StrictMode>
          <Provider store={store}>
            <Router>
              <AddRoom />
            </Router>
          </Provider>
          ,
        </React.StrictMode>,
      );
      expect(addRoom).toMatchSnapshot();
    });
  });