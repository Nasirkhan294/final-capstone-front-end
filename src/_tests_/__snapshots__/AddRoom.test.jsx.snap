// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRoom component Should render the proper AddRoom component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative flex flex-col bg-clip-border rounded-xl text-gray-700 shadow-md mt-8 mb-64 max-w-[450px] mx-auto bg-white/90 backdrop-blur-md"
      >
        <div
          class="relative bg-clip-border mx-4 rounded-xl overflow-hidden shadow-lg -mt-6 mb-4 grid h-28 place-items-center bg-orange-600/90 text-white backdrop-blur-md"
        >
          <h3
            class="block antialiased text-3xl leading-snug text-white font-osans uppercase tracking-widest font-light"
          >
            Add a room
          </h3>
        </div>
        <div
          class="p-6 text-center"
        >
          <h5
            class="block antialiased tracking-normal font-sans text-xl font-semibold leading-snug text-inherit mb-2"
          >
            Kind Note
          </h5>
          <p
            class="block antialiased font-sans text-base font-light leading-relaxed text-inherit text-center"
          >
            Kindly fill in the form below to add a room to the list of available rooms.
          </p>
        </div>
        <form
          action="#"
        >
          <div
            class="p-6 flex flex-col gap-4"
          >
            <div>
              <img
                alt="room-preview"
                src="https://www.fluttercampus.com/img/4by3.webp"
              />
            </div>
            <div
              class="relative w-full min-w-[200px] h-10"
            >
              <input
                class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-2.5 rounded-[7px] border-blue-gray-200 focus:border-orange-500"
                name="name"
                placeholder=" "
                value=""
              />
              <label
                class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal !overflow-visible truncate peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[3.75] text-blue-gray-400 peer-focus:text-orange-500 before:border-blue-gray-200 peer-focus:before:!border-orange-500 after:border-blue-gray-200 peer-focus:after:!border-orange-500"
              >
                Name
                 
              </label>
            </div>
            <div
              class="relative w-full min-w-[200px] h-10"
            >
              <input
                class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-2.5 rounded-[7px] border-blue-gray-200 focus:border-orange-500"
                name="image"
                placeholder=" "
                value="https://www.fluttercampus.com/img/4by3.webp"
              />
              <label
                class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal !overflow-visible truncate peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[3.75] text-blue-gray-400 peer-focus:text-orange-500 before:border-blue-gray-200 peer-focus:before:!border-orange-500 after:border-blue-gray-200 peer-focus:after:!border-orange-500"
              >
                Put a Room Image link
                 
              </label>
            </div>
            <div
              class="relative w-full min-w-[200px] h-11"
            >
              <input
                class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-3 rounded-md border-blue-gray-200 focus:border-orange-500"
                name="price"
                placeholder=" "
                type="number"
                value=""
              />
              <label
                class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal !overflow-visible truncate peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[4.1] text-blue-gray-400 peer-focus:text-orange-500 before:border-blue-gray-200 peer-focus:before:!border-orange-500 after:border-blue-gray-200 peer-focus:after:!border-orange-500"
              >
                Price
                 
              </label>
            </div>
            <div
              class="relative w-full min-w-[200px]"
            >
              <textarea
                class="peer w-full h-full min-h-[100px] bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 resize-y disabled:bg-blue-gray-50 disabled:border-0 disabled:resize-none transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-2.5 rounded-[7px] border-blue-gray-200 focus:border-orange-500 !resize-none"
                name="description"
                placeholder=" "
              />
              <label
                class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[3.75] text-blue-gray-400 peer-focus:text-orange-500 before:border-blue-gray-200 peer-focus:before:!border-orange-500 after:border-blue-gray-200 peer-focus:after:!border-orange-500"
              >
                Description
              </label>
            </div>
            <div
              class="inline-flex items-center"
            >
              <label
                class="relative flex items-center cursor-pointer p-3 rounded-full"
                for=":r1:"
              >
                <input
                  checked=""
                  class="peer relative appearance-none w-5 h-5 border rounded-md border-blue-gray-200 cursor-pointer transition-all before:content[''] before:block before:bg-blue-gray-500 before:w-12 before:h-12 before:rounded-full before:absolute before:top-2/4 before:left-2/4 before:-translate-y-2/4 before:-translate-x-2/4 before:opacity-0 hover:before:opacity-10 before:transition-opacity checked:bg-orange-500 checked:border-orange-500 checked:before:bg-orange-500"
                  id=":r1:"
                  name="available"
                  type="checkbox"
                  value="true"
                />
                <span
                  class="text-white absolute top-2/4 left-2/4 -translate-y-2/4 -translate-x-2/4 pointer-events-none opacity-0 peer-checked:opacity-100 transition-opacity"
                >
                  <svg
                    class="h-3.5 w-3.5"
                    fill="currentColor"
                    stroke="currentColor"
                    stroke-width="1"
                    viewBox="0 0 20 20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </label>
              <label
                class="text-gray-700 font-light select-none cursor-pointer mt-px"
                for=":r1:"
              >
                Available
              </label>
            </div>
          </div>
          <div
            class="p-6 pt-0"
          >
            <button
              class="align-middle select-none font-sans font-bold text-center transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs py-3 px-6 rounded-lg bg-gradient-to-tr from-orange-600 to-orange-400 text-white shadow-md shadow-orange-500/20 hover:shadow-lg hover:shadow-orange-500/40 active:opacity-[0.85] w-full capitalize flex justify-center items-center"
              disabled=""
              type="submit"
            >
              <span>
                Add Room
              </span>
            </button>
          </div>
        </form>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="relative flex flex-col bg-clip-border rounded-xl text-gray-700 shadow-md mt-8 mb-64 max-w-[450px] mx-auto bg-white/90 backdrop-blur-md"
    >
      <div
        class="relative bg-clip-border mx-4 rounded-xl overflow-hidden shadow-lg -mt-6 mb-4 grid h-28 place-items-center bg-orange-600/90 text-white backdrop-blur-md"
      >
        <h3
          class="block antialiased text-3xl leading-snug text-white font-osans uppercase tracking-widest font-light"
        >
          Add a room
        </h3>
      </div>
      <div
        class="p-6 text-center"
      >
        <h5
          class="block antialiased tracking-normal font-sans text-xl font-semibold leading-snug text-inherit mb-2"
        >
          Kind Note
        </h5>
        <p
          class="block antialiased font-sans text-base font-light leading-relaxed text-inherit text-center"
        >
          Kindly fill in the form below to add a room to the list of available rooms.
        </p>
      </div>
      <form
        action="#"
      >
        <div
          class="p-6 flex flex-col gap-4"
        >
          <div>
            <img
              alt="room-preview"
              src="https://www.fluttercampus.com/img/4by3.webp"
            />
          </div>
          <div
            class="relative w-full min-w-[200px] h-10"
          >
            <input
              class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-2.5 rounded-[7px] border-blue-gray-200 focus:border-orange-500"
              name="name"
              placeholder=" "
              value=""
            />
            <label
              class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal !overflow-visible truncate peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[3.75] text-blue-gray-400 peer-focus:text-orange-500 before:border-blue-gray-200 peer-focus:before:!border-orange-500 after:border-blue-gray-200 peer-focus:after:!border-orange-500"
            >
              Name
               
            </label>
          </div>
          <div
            class="relative w-full min-w-[200px] h-10"
          >
            <input
              class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-2.5 rounded-[7px] border-blue-gray-200 focus:border-orange-500"
              name="image"
              placeholder=" "
              value="https://www.fluttercampus.com/img/4by3.webp"
            />
            <label
              class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal !overflow-visible truncate peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[3.75] text-blue-gray-400 peer-focus:text-orange-500 before:border-blue-gray-200 peer-focus:before:!border-orange-500 after:border-blue-gray-200 peer-focus:after:!border-orange-500"
            >
              Put a Room Image link
               
            </label>
          </div>
          <div
            class="relative w-full min-w-[200px] h-11"
          >
            <input
              class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-3 rounded-md border-blue-gray-200 focus:border-orange-500"
              name="price"
              placeholder=" "
              type="number"
              value=""
            />
            <label
              class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal !overflow-visible truncate peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[4.1] text-blue-gray-400 peer-focus:text-orange-500 before:border-blue-gray-200 peer-focus:before:!border-orange-500 after:border-blue-gray-200 peer-focus:after:!border-orange-500"
            >
              Price
               
            </label>
          </div>
          <div
            class="relative w-full min-w-[200px]"
          >
            <textarea
              class="peer w-full h-full min-h-[100px] bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 resize-y disabled:bg-blue-gray-50 disabled:border-0 disabled:resize-none transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-2.5 rounded-[7px] border-blue-gray-200 focus:border-orange-500 !resize-none"
              name="description"
              placeholder=" "
            />
            <label
              class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[3.75] text-blue-gray-400 peer-focus:text-orange-500 before:border-blue-gray-200 peer-focus:before:!border-orange-500 after:border-blue-gray-200 peer-focus:after:!border-orange-500"
            >
              Description
            </label>
          </div>
          <div
            class="inline-flex items-center"
          >
            <label
              class="relative flex items-center cursor-pointer p-3 rounded-full"
              for=":r1:"
            >
              <input
                checked=""
                class="peer relative appearance-none w-5 h-5 border rounded-md border-blue-gray-200 cursor-pointer transition-all before:content[''] before:block before:bg-blue-gray-500 before:w-12 before:h-12 before:rounded-full before:absolute before:top-2/4 before:left-2/4 before:-translate-y-2/4 before:-translate-x-2/4 before:opacity-0 hover:before:opacity-10 before:transition-opacity checked:bg-orange-500 checked:border-orange-500 checked:before:bg-orange-500"
                id=":r1:"
                name="available"
                type="checkbox"
                value="true"
              />
              <span
                class="text-white absolute top-2/4 left-2/4 -translate-y-2/4 -translate-x-2/4 pointer-events-none opacity-0 peer-checked:opacity-100 transition-opacity"
              >
                <svg
                  class="h-3.5 w-3.5"
                  fill="currentColor"
                  stroke="currentColor"
                  stroke-width="1"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </label>
            <label
              class="text-gray-700 font-light select-none cursor-pointer mt-px"
              for=":r1:"
            >
              Available
            </label>
          </div>
        </div>
        <div
          class="p-6 pt-0"
        >
          <button
            class="align-middle select-none font-sans font-bold text-center transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs py-3 px-6 rounded-lg bg-gradient-to-tr from-orange-600 to-orange-400 text-white shadow-md shadow-orange-500/20 hover:shadow-lg hover:shadow-orange-500/40 active:opacity-[0.85] w-full capitalize flex justify-center items-center"
            disabled=""
            type="submit"
          >
            <span>
              Add Room
            </span>
          </button>
        </div>
      </form>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
